### ISSUE TITLE
[Request] [Idea] Flutter Performance Lints

### ISSUE THREAD

**Issue by ghost** [dart-lang/sdk#58403]
*@henry2man commented on May 14, 2021, 6:07 AM UTC:*

TL;DR:

_As a developer -sometimes- I introduce performance problems in my applications_ <g-emoji class="g-emoji" alias="sweat_smile" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f605.png">😅</g-emoji> **What if the Flutter plugin helps me detect and fix them?**

Full story:

Reviewing some articles on recommendations to improve the performance of applications made with Flutter, I find some elements that could be added to the plugin to help developers improve the performance of their apps.

For example:

*   It is recommended not to use methods that create Widgets (or lists of Widgets) within the `build()` method. Instead, specific Widgets must be created ([https://www.youtube.com/watch?v=vVg9It7cOfY&t=1320s](https://www.youtube.com/watch?v=vVg9It7cOfY&t=1320s), [https://blog.codemagic.io/how-to-improve-the-performance-of-your](https://blog.codemagic.io/how-to-improve-the-performance-of-your) -flutter-app./#dont-split-your-widgets-into-methods)
*   It is recommended to use lazy lists instead of creating all items manually ([https://flutter.dev/docs/perf/rendering/best-practices#render-grids-and-lists-lazily](https://flutter.dev/docs/perf/rendering/best-practices#render-grids-and-lists-lazily))
*   It is recommended not to use the `Opacity` widget ([https://flutter.dev/docs/perf/rendering/best-practices#pitfalls](https://flutter.dev/docs/perf/rendering/best-practices#pitfalls))

I am not an absolute specialist in performance with Flutter but I have debugged on other platforms and it is a very important element in mobile apps.

My suggestion is that some of these items could appear as part of the warnings and/or as part of a specific performance analysis extension within the plugin.

What do you think?

*This issue was moved by [pq](https://github.com/pq) from [flutter/flutter-intellij#5484](https://github.com/flutter/flutter-intellij/issues/5484).*

**Comment by srawlins** [comment]
We won't be implementing this as a warning or a lint rule in the Dart analyzer. We now have a high bar for implementing and maintaining built-in static analysis rules, and these rules are not specified well enough, and may not be a fit for static analysis, or have high enough impact. As of Dart 3.10 (Flutter 3.38), a [new analyzer plugins system](https://dart.dev/tools/analyzer-plugins) is available, and the requested static analysis here could be implemented as an analyzer plugin.
